torchquantum/encoding/encodings.py

```diff
@@ -208,41 +250,52 @@ def forward(self, qdev: tq.QuantumDevice, x):
 class StateEncoder(Encoder, metaclass=ABCMeta):
     """StateEncoder is a subclass of Encoder and represents a state encoder.
     It encodes the input data into the state vector of a quantum device."""
+
     def __init__(self):
         super().__init__()
 
     def forward(self, qdev: tq.QuantumDevice, x):
         """
-            Performs the encoding by preparing the state vector of the quantum device.
+        Performs the encoding by preparing the state vector of the quantum device.
 
-                Args:
-                    qdev (tq.QuantumDevice): The quantum device to be used for encoding.
-                    x (torch.Tensor): The input data to be encoded.
-                Returns:
-                    torch.Tensor: The encoded data.
+            Args:
+                qdev (tq.QuantumDevice): The quantum device to be used for encoding.
+                x (torch.Tensor): The input data to be encoded.
+            Returns:
+                torch.Tensor: The encoded data.
 
-                """
+        """
+        # Validate inputs
+        self.validate_inputs(qdev, x)
```

Comment:
Want to check if we want this @Hanrui-Wang. The pro is that it’s safer from errors but also could result in some extra overhead.